src/integrationTest/java/uk/gov/hmcts/reform/iacasedocumentsapi/component/testutils/SpringBootIntegrationTest.java

```diff
@@ -2,84 +2,117 @@
+    private Resource resourceOpenIdConfigurationFile;
+
+    protected GivensBuilder given;
+
+    @LocalServerPort
+    protected int port;
+
     @Rule
     public WireMockRule wireMockRule = new WireMockRule(8990);
 
-    private MockMvc mockMvc;
+    protected IaCaseDocumentsApiClient iaCaseDocumentsApiClient;
+    protected DocmosisStub docmosisStub = new DocmosisStub();
 
-    @Autowired
-    private WebApplicationContext webApplicationContext;
+    @Before
+    public void setUpGivens() {
+        given = new GivensBuilder();
+    }
 
-    protected IaApiClient iaApiClient;
-    protected DocmosisStub docmosisStub = new DocmosisStub();
+    @Before
+    public void setUpApiClient() {
+        iaCaseDocumentsApiClient = new IaCaseDocumentsApiClient(port);
+    }
 
     @Before
     public void setUpTestInfrastructure() {
-        this.mockMvc = webAppContextSetup(webApplicationContext).build();
-        this.iaApiClient = new IaApiClient(mockMvc);
         this.docmosisStub.withDefaults();
     }
 
     @Before
-    public void setupIdamStubs() {
+    public void setupIdamStubs() throws IOException {
 
-        stubFor(get(urlEqualTo("/details"))
-                .willReturn(aResponse()
-                        .withStatus(200)
-                        .withHeader("Content-Type", "application/json")
-                        .withBody(AsylumCaseFixtures.someUserDetails())));
+        stubFor(get(urlEqualTo("/userAuth/o/.well-known/openid-configuration"))
+            .willReturn(aResponse()
+                .withStatus(200)
+                .withHeader("Content-Type", "application/json")
+                .withBody(FileUtils.readFileToString(resourceOpenIdConfigurationFile.getFile()))));
+
+        stubFor(get(urlEqualTo("/userAuth/o/jwks"))
+            .willReturn(aResponse()
+                .withStatus(200)
+                .withHeader("Content-Type", "application/json")
+                .withBody(FileUtils.readFileToString(resourceJwksFile.getFile()))));
+
+        stubFor(get(urlEqualTo("/userAuth/o/userinfo"))
+            .willReturn(aResponse()
+                .withStatus(200)
+                .withHeader("Content-Type", "application/json")
+                .withBody(AsylumCaseFixtures.someUserDetails())));
     }
 
     @Before
     public void setupDocumentUploadStub() {
 
         stubFor(post(urlEqualTo("/documents"))
-                .withMultipartRequestBody(aMultipart().matchingType(MatchingType.ALL))
-                .willReturn(aResponse()
-                        .withStatus(201)
-                        .withBody(someUploadResponse())));
+            .withMultipartRequestBody(aMultipart().matchingType(MatchingType.ALL))
+            .willReturn(aResponse()
+                .withStatus(201)
+                .withBody(someUploadResponse())));
     }
 
     @Before
     public void setupS2S() {
 
-        stubFor(post(urlEqualTo("/lease"))
+        stubFor(get(urlEqualTo("/serviceAuth/details"))
```

Comment:
Did s2s change also?

Reply:
It was changed from `AuthCheckerServiceAndUserFilter` to `AuthCheckerServiceOnlyFilter` but the logic is the same.
The reason for that stub is that I refactored tests to use `RestTemplate` client instead of `MockMvc`.
`MockMvc` didn't use auth filter so s2s wasn't checked at all and user validation was checked in controller / dispatcher logic.
Currently

Reply:
👍